mmocr/datasets/preparers/obtainers/aws_s3_obtainer.py

```diff
@@ -0,0 +1,237 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import glob
+import os
+import os.path as osp
+import shutil
+import ssl
+from typing import Dict, List, Optional, Tuple
+
+import boto3
+from botocore import UNSIGNED
+from botocore.config import Config
```

Comment:
Since not everyone needs the s3 client, we can turn this requirement as optional by moving the import util into `AWSS3Obtainer.__init__` where these packages are actually needed. Also, try to give a proper error msg in case of missing package like:

https://github.com/open-mmlab/mmocr/blob/37c5d371c7cbadfcde58af20206bf7b544d9f329/mmocr/datasets/recog_lmdb_dataset.py#L158-L162